### ISSUE TITLE
BUG: Incorrect Assembly Engineer Turret Behavior

### ISSUE THREAD

**Issue by bhahouighf** [DarkflameUniverse/DarkflameServer#603]
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

Server version commit: 94e161df55ed3bf28e4c71c0758299540247ba84

### Issue Description

At present, the Assembly Engineer Rank 2 turret outperforms the Rank 3 turret in DPS. The Rank 3 turret has a rate of fire faster than that of the Rank 1 turret but slower than that of the Rank 2 turret, meaning the Rank 2 and Rank 3 behaviors are switched. The Rank 2 turret currently fires at around one shot per half-second, which, while it cannot be verified against any live footage I know of, aligns with information about the Rank 3 turret on the LU wiki circa 2012.

![image](https://user-images.githubusercontent.com/108308609/176055901-f475fedd-d424-434d-9eaa-ffb5af702ea4.png)

https://user-images.githubusercontent.com/108308609/176054492-f66cd349-09ae-45be-9a4e-e18bbf4decd8.mp4

https://user-images.githubusercontent.com/108308609/176054347-90d7d1ff-b6d0-4529-b7ef-ede5bd56bed6.mp4

https://user-images.githubusercontent.com/108308609/176054322-d206cf03-4c5b-4c5b-b7fc-7456777dba12.mp4

### Reproduction Steps

Spawn an Engineer Rank 2 and/or Rank 3 turret.

### Expected Behavior

Rank 3 turrets should exhibit the behavior currently shown by Rank 2 turrets, and vice versa.

### Environment

Client: Windows 10 + unpacked client
Server: Google Cloud, Debian 11

**Comment by EmosewaMC** [comment]
This is not a bug.  We are simply reading in the data that the turrets behavior has.  Feel free to buff this on your own server or show proof from live that this is a bug but we will not be fixing this.  Closing.

**Comment by EmosewaMC** [comment]
An update to this - after some further digging, we are potentially leaning towards this being a simple data entry error that happened in live.  Still triaging the issue.

**Comment by Marcono1234** [comment]
Here are two videos from live:
- https://youtu.be/H_FJDPAcNd4?t=191
from 2010, so maybe too old to be relevant
- https://youtu.be/B3qB1YZVMAA?t=170
does not show turret that well, but there are definitely shots which are less than half a second apart (could that be affected by lag?)

**Comment by EmosewaMC** [comment]
> Here are two videos from live:
> 
> * https://youtu.be/H_FJDPAcNd4?t=191
>   from 2010, so maybe too old to be relevant
> * https://youtu.be/B3qB1YZVMAA?t=170
>   does not show turret that well, but there are definitely shots which are less than half a second apart (could that be affected by lag?)

Lag would not be an issue.  Combat is client calculated and server confirmed.  These videos are not good enough proof to warrant a fix to main.  Please provide clearer proof.

**Comment by heatraydeathgun** [comment]
I found a few videos that perhaps have sufficient time with the assembly turret in play. I also made a few casual calculations of shots per second for both the videos in this issue and the ones from the videos.

Video 1: From 1-6s (5s firing) we have 5 shots for 1shot/s
Metronome approximately matches 45bpm (0.75bps)
Video 2: From 4-10s (6s firing) we have 10 shots for 1.6667shots/s 
Metronome approximately matches 90bpm (1.6667bps)
Video 3: From 4-16s (12s firing) we have 15 shots for 1.25shots/s 
Metronome approximately matches 1.16667

Mar 2011: https://www.youtube.com/watch?v=fHOwW5JkalIt=47
From 47-50s (3s firing) we have 5 shots = 1.6667shots/s
Metronome approximately matches at 100bpm (1.6667bps)

Dec 2011: (https://youtu.be/920rf6HrF_k?t=759)
From 12:42-12:45s (4s firing) we have 7 shots = 1.75shots/s 
Metronome approximately matches at 100bpm (1.6667bps)

Feb 2011: https://www.youtube.com/watch?v=QuF4mJf8qPc?t=122
From 30-50s (20s) we have 37 shots = 1.85shots/s 
From 2:02-2:21s (19s) we have 34 shots = 1.79shots/s 
Metronome approximately matches at 114bpm (1.9bps)

Given the camera work and my imprecise timekeeping there's definitely a bit of error here. The February video has a slightly faster rate of fire, and the video doesn't appear to be sped up (would be 1.14x), so perhaps there was a change between when it was recorded and the next video. I also haven't found a clip of Rank1 or Rank2 for comparison, but these videos and the matching to a metronome at least indicate that the live speed was faster than what we see currently in the Rank3 turret.
